src/check_datapackage/check.py

```diff
@@ -48,17 +48,40 @@ def __init__(
         self,
         issues: list[Issue],
     ) -> None:
-        """Create the DataPackageError attributes from issues."""
-        # TODO: Switch to using `explain()` once implemented
-        errors: list[str] = _map(
-            issues,
-            lambda issue: f"- Property `{issue.jsonpath}`: {issue.message}\n",
-        )
-        message: str = (
-            "There were some issues found in your `datapackage.json`:\n\n"
-            + "\n".join(errors)
-        )
-        super().__init__(message)
+        """Create the DataPackageError from issues."""
+        super().__init__(explain(issues))
+
+
+def explain(issues: list[Issue]) -> str:
+    """Explain the issue in a human-readable format.
+
+    Args:
+        issues: A list of `Issue` objects to explain.
+
+    Returns:
+        A human-readable explanation of the issue.
+    """
+    issue_explanations: list[str] = _map(
+        issues,
+        _create_explanation,
+    )
+    error_message: str = (
+        f"{len(issue_explanations)} issues were found in your `datapackage.json`:\n\n"
+        + "\n".join(issue_explanations)
+    )
+    return error_message
+
+
+def _create_explanation(issue: Issue) -> str:
+    """Create an informative explanation of what went wrong in each issue."""
+    property_name = issue.jsonpath.split(".")[-1]
+    return (  # noqa: F401
+        f"At package.{issue.jsonpath[2:]}:\n"  # indexing to remove '$.'
```

Comment:
```suggestion
        f"At `package.{issue.jsonpath[2:]}`:\n"  # indexing to remove '$.'
```

I think backticks or some other visual separator would be nice here.

Reply:
Personally, I think it can look a bit cluttered with raw markup being printed as output. Could we punt this to a PR that deals with the styles of these messages and considers using rich? E.g. just using the `print` from rich would give us some additional formatting:

<img width="450" height="274" alt="image" src="https://github.com/user-attachments/assets/8c51d308-836a-46dd-b467-30173b001529" />


Rich also does pretty printing of json objects and creates stylized tracebacks, such as this example:

<img width="926" height="369" alt="image" src="https://github.com/user-attachments/assets/d59000a3-4d5a-488a-90d0-0efc3f2517fb" />